cmd/podman/images/scp.go

```diff
@@ -0,0 +1,383 @@
+			}
+			dedup[fp] = s
+		}
+		var uniq []ssh.Signer
+		for _, s := range dedup {
+			uniq = append(uniq, s)
+		}
+		authMethods = append(authMethods, ssh.PublicKeysCallback(func() ([]ssh.Signer, error) {
+			return uniq, nil
+		}))
+	}
+	if passwdSet { // proper password validation for ssh connection if necessary
+		authMethods = append(authMethods, ssh.Password(passwd))
+	}
+	if len(authMethods) == 0 {
+		authMethods = append(authMethods, ssh.PasswordCallback(func() (string, error) {
+			pass, err := t.ReadPassword(fmt.Sprintf("%s's login password:", uri.User.Username()))
+			return string(pass), err
+		}))
+	}
+	cfg := &ssh.ClientConfig{ // configure ssh client
+		User:            uri.User.Username(),
+		Auth:            authMethods,
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+	}
+	dial, err := ssh.Dial("tcp", uri.Host, cfg) // dial the client
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to connect")
+	}
+	defer dial.Close()
+
+	session, err := dial.NewSession() // connect to the dialed client
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to create new ssh session on %q", uri.Host)
+	}
+	defer session.Close()
+	if err != nil {
+		return "", err
+	}
+	if ToFrom { // if we are loading to the remote/ssh client
+		f, _ := os.Open(imageFileIn) // open image file to copy
+		defer f.Close()
+		n, err := scpD.CopyTo(dial, imageFileIn, imageFileOut)
+		if err != nil {
+			fmt.Println(n, "Bytes copied before error. Error while copying file ", err)
+		}
+
+		podman := "podman"
+		if v, found := os.LookupEnv("PODMAN_BINARY"); found {
+			podman = v // get path to podman bin
+		}
+		run := podman + " image load --input=" + imageFileOut // run ssh image load of the file copied via scp
+		var buffer bytes.Buffer
+		session.Stdout = &buffer
+		if err := session.Run(run); err != nil { // run the command declared above
+			return "", err
+		}
+		return buffer.String(), nil // return the output of image load on the machine, should be "Loaded Image(s)..."
+	} // else we are loading from the ssh client
+	podman := "podman"
+	if v, found := os.LookupEnv("PODMAN_BINARY"); found {
+		podman = v // get path to podman bin
+	}
+	run := podman + " image save " + imageFileOut + " --format=oci-archive --output=" + imageFileIn // run ssh image load of the file copied via scp. Files are reverse in thie case...
+	var buffer bytes.Buffer
+	session.Stdout = &buffer
+
+	if err := session.Run(run); err != nil { // run the command declared above
+		return "", err
+	}
+	n, err := scpD.CopyFrom(dial, imageFileIn, imageFileIn)
+	if err != nil {
+		fmt.Print(n, " bytes copied before err")
+		return "", err
+	}
+	return "", nil
+}
+
+// getUserInfo takes a ssh url and extracts user information
+func getUserInfo(uri *urlP.URL) (*urlP.Userinfo, error) {
+	var (
+		usr *user.User
+		err error
+	)
+	if u, found := os.LookupEnv("_CONTAINERS_ROOTLESS_UID"); found {
```

Comment:
Should `image scp` be hard-coding things like this? Another part that looks like something that should be a shared ssh infrastructure.